tests/unit/test_loki_server.py

```diff
@@ -48,7 +31,59 @@ def mocked_requests_get_exception(*args, **kwargs):
     raise requests.exceptions.HTTPError("404")
 
 
+class ThreadWithStop(threading.Thread):
+    def __init__(self, *args, **kwargs):
+        super(ThreadWithStop, self).__init__(*args, **kwargs)
+        self.stop = threading.Event()
+
+    def run(self):
+        while True:
+            self.stop.wait()
+
+
+class FakeLokiRequestHandler(SimpleHTTPRequestHandler):
```

Comment:
Does this test charm code better than a mock?

Reply:
It's more or less required to verify that `requests` does not spam logs. `requests` must be _actuallly_ invoked.

As an additional comment, removing the huge amount of mocking in the unit tests is an overall goal so we can test the "actual" workflow.